### PR TITLE
vtk: update to 9.1.0

### PR DESCRIPTION
**Pull request by mdave**
#### Description

This PR bumps the version of vtk to 9.1.0 (November 2021). Changes made:

- `CMAKE_BUILD_TYPE` was changed from `MacPorts` to `Release` owing to an issue in the building of Eigen3 inside VTK (using `MacPorts` gives an 'invalid build type' error).
- `PYTHON_EXECUTABLE` et al are now deprecated in favour of `Python3_EXECUTABLE` et al.
- Similarly, `VTK_INSTALL_PYTHON_MODULES_DIR` no longer exists; this has been deprecated in favour of `VTK_PYTHON_SITE_PACKAGES_SUFFIX`, but means I have hard-coded a `Library/Frameworks` relative path instead of using `${frameworks_path}`.
- Removed various patches which I think have been upstreamed and are no longer needed?

There didn't appear to be many pots that depend directly on VTK for me to test this against, but my own app builds and executes against this version okay. I also don't know if there are other things to be aware of, but thought this might get the conversation started on updating VTK at least!

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on
macOS 12.4 21F79 x86_64
Xcode 13.4.1 13F100

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: graphics/vtk/Portfile
Comment (mascguy) [inline graphics/vtk/Portfile:18]: You'll want to change `revision` to zero.

**Comment by rdbisme** [comment]
Can't test this right now because don't have an Apple machine at the moment, but I plan to fix this :). The rest looks ok.

**Comment by mascguy** [comment]
> Can't test this right now because don't have an Apple machine at the moment, but I plan to fix this :). The rest looks ok. 
> Thanks! 

Great, I'll merge once the revision has been fixed. Thanks for following up!

File: graphics/vtk/Portfile
Comment (mdave) [reply graphics/vtk/Portfile]: Oops! Have changed to 0, rebased to current master and squashed.

**Review by mascguy — APPROVED** [review]
LGTM, and maintainer approval provided. Will merge once CI completes.

**Pull request merged**
